pytest_plugins/infra_dependent_markers.py

```diff
@@ -0,0 +1,20 @@
+# Custom infra dependent markers for selection and deselection of tests
+
+
+def pytest_configure(config):
+    """Register custom markers to avoid warnings."""
+    markers = []
+    markers.extend(infra_markers())
+    for marker in markers:
+        config.addinivalue_line("markers", marker)
+
+
+def infra_markers():
+    """ Register custom markers for infra dependent tests """
+    return [
+        "on_premises_provisioning: Tests that runs on on_premises Providers",
+        "libvirt_discovery: Tests depends on Libvirt Provider for discovery",
+        "libvirt_content_host: Tests depends on Libvirt Provider for content hosts",
```

Comment:
In SatLab  there is only RHVM and OSP providers, I dont like names: 
`libvirt_*`

Reply:
I'm OK with:
- `discovery`
- `content_host`

Reply:
Right, The intuition of this PR is to filter the tests to exclude in SatLab :) and hence they are explicitly called with `libvirt_*`.

Reply:
@lpramuk They are specially marked with `libvirt_` because discovery and content_host can also be the hosts of other compute resources like VMWare / RHEV etc, but these libvirt based tests are explicitly asking for old infra libvirt and hence we ll exclude only `libvirt_content_host` but not `rhev_content_host`.

I dont want put all of them in generic category of `content_host` so that excluding them would get difficult.

Reply:
@lpramuk correct, many of these libvirt tests will get removed and replaced by compute resource tests against compute resources that are officially supported (libvirt is not, but was used for testing convenience).

We want all of these libvirt tests marked and deselected by default so that they no longer run in the new pipelines.